pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -509,6 +509,189 @@ def pow(self, z):
         return [PauliRot(self.data[0] * z, self.hyperparameters["pauli_word"], wires=self.wires)]
 
 
+class PCPhase(Operation):
+    r"""
+    A projector-controlled phase gate.
+
+    .. math:: \Pi_{\phi}(\phi, dim=2, wires=[0,1]) = \begin{bmatrix}
+                e^{i\phi} & 0 & 0 & 0 \\
+                0 & e^{i\phi} & 0 & 0 \\
+                0 & 0 & 1 & 0 \\
+                0 & 0 & 0 & 1
+            \end{bmatrix}.
```

Comment:
I have the same question for the `BlockEncode` operation. I know we try to keep the descriptions for operations short.
But as a first time user I would struggle to understand how to use `dim` and how this is "projector-controlled."
Would it be useful to add more information or link to a paper?

Reply:
Added a paragraph explaining this !